### PR TITLE
Make docs do less work/be less flaky

### PR DESCRIPTION
**Pull request by dnfield**
This should make our docs step go faster on post submit, by requiring us to send fewer files to firebase hosting - because fewer files will change.  That will only be true after this runs at least once - after that, we will be able to send just the files that change.

This is a bit hacky, but could be made less so by https://github.com/dart-lang/dartdoc/issues/1982

/cc @gspencergoog @jcollins-g

### CODE REVIEW

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -85,7 +92,7 @@ Future<void> main(List<String> arguments) async {
 
   final String pubExecutable = '$flutterRoot/bin/cache/dart-sdk/bin/pub';
 
-  // Run pub.
+  // // Run pub.
```
Comment (jonahwilliams) [inline dev/tools/dartdoc.dart]:
```suggestion
  // Run pub.
```

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (tvolkert) [inline dev/tools/dartdoc.dart]: Extra `//`

File: dev/tools/dartdoc.dart [hunk 2]
```diff
@@ -50,7 +50,14 @@ Future<void> main(List<String> arguments) async {
   final StringBuffer buf = StringBuffer();
   buf.writeln('name: Flutter');
   buf.writeln('homepage: https://flutter.dev');
-  buf.writeln('version: $version');
+  // TODO(dnfield): We should make DartDoc able to avoid emitting this. If we
+  // use the real value here, every file will get marked as new instead of only
+  // files that have otherwise changed. Instead, we replace it dynamically using
+  // JavaScript.
+  // We'll replace this dynamically. This way, fewer files will get marked as
```
Comment (tvolkert) [inline dev/tools/dartdoc.dart]: "We'll replace this dynamically" is redundant here (already said in the previous sentence)

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**